### PR TITLE
kazel: skip third_party/etcd.*

### PR DESCRIPTION
**Pull request by ixdy**
**What this PR does / why we need it**: fixes the issue encountered in https://github.com/kubernetes/kubernetes/pull/62151#discussion_r179322174.

Basically, we git ignore anything under `third_party/etcd.*`, but if a `BUILD` file ends up in there somehow, it'll confuse kazel and generate invalid entries in top-level `BUILD` files that we do check in. We can avoid this issue by instructing kazel to ignore these directories.

**Release note**:

```release-note
NONE
```

/assign @mikedanese @thockin @BenTheElder

### CODE REVIEW

**Review by BenTheElder — APPROVED** [review]
/lgtm
/hold

**Comment by ixdy** [comment]
can I
/approve
this now?

**Comment by ixdy** [comment]
@cjwagner After https://github.com/kubernetes/kubernetes/pull/62484 I'd expect to have approval rights here now. Did I mess this up somehow?

**Comment by ixdy** [comment]
let's try that again...

/approve
?

**Comment by cjwagner** [comment]
@ixdy This isn't a problem with the OWNERS file regexp filtering itself, but the `approve` plugin's use of the `repoowners` package. The `approve` plugin isn't respecting the interface exposed by `RepoOwners` and is making a bunch of invalid assumptions: https://github.com/kubernetes/test-infra/issues/7690
Unfortunately, I think it will require a significant rewrite. In the mean time regexp filtering will only work for  other plugins like `blunderbuss` and `owners-label`.

**Pull request merged**
